janus_server/src/aggregator/aggregate_share.rs

```diff
@@ -207,6 +207,19 @@ impl CollectJobDriver {
 
         Ok(())
     }
+
+    #[tracing::instrument(skip(self, datastore), err)]
+    pub async fn cancel_collect_job<C: Clock>(
+        &self,
+        datastore: Arc<Datastore<C>>,
+        lease: Lease<AcquiredCollectJob>,
+    ) -> Result<(), Error> {
+        let collect_job_id = lease.leased().collect_job_id;
+        datastore
+            .run_tx(|tx| Box::pin(async move { tx.cancel_collect_job(collect_job_id).await }))
```

Comment:
I need to also release the lease in this transaction (missed this on the initial implementation, whoops).